src/stakes/page.rs

```diff
@@ -37,35 +34,38 @@ pub fn StakesPage() -> impl IntoView {
     let (current_page, set_current_page) = create_signal(1);
 
     view! {
+        <Title
+            text=move || {
+                if let Some(epoch) = epoch_sig.get() {
+                    format!("Epoch {}", epoch)
+                } else {
+                    "Current Staking Ledger".to_string()
+                }
+            }
+
+            formatter=move |text| format!("Staking Ledger | {text}")
+        />
         <SearchBar placeholder="Exact search for public key".to_string()/>
         <PageContainer>
             {move || match resource.get() {
                 Some(Ok(data)) => {
                     let (previous_epoch, next_epoch, curr_epoch, section_heading) = match (
                         current_epoch(),
-                        query_string_epoch(),
+                        epoch_sig.get(),
                     ) {
                         (Some(curr_epoch), Some(qs_epoch)) => {
-                            let i_curr_epoch = curr_epoch as i64;
-                            match qs_epoch.parse::<i64>() {
-                                Ok(i_qs_epoch) => {
-                                    let i_next_epoch = min(i_curr_epoch, i_qs_epoch + 1);
-                                    let i_prev_epoch = max(0, i_qs_epoch - 1);
-                                    let header = if i_next_epoch == i_qs_epoch {
-                                        "Current Staking Ledger".to_string()
-                                    } else {
-                                        format!("Epoch {} Staking Ledger", i_qs_epoch)
-                                    };
-                                    (i_prev_epoch, i_next_epoch, i_curr_epoch, header)
-                                }
-                                _ => (0, 0, 0, "".to_string()),
-                            }
+                            let header = if curr_epoch == qs_epoch {
```

Comment:
This is unrelated, no?

Reply:
This one is related. The test revealed this logic wasn't working quite right.